grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -149,28 +149,15 @@ class LocationService {
             }
         }
 
-        def outboundMovementLocations = locations.findAll {
-            (it.locationGroup == currentLocation.locationGroup) ||
-                    (it.locationGroup != currentLocation.locationGroup && it.locationType.locationTypeCode == LocationTypeCode.DEPOT)
-        }
-
-        if (!isSuperuser) {
-            if (direction == "INBOUND") {
-                return locations.findAll {
-                    it.locationType.locationTypeCode == LocationTypeCode.SUPPLIER
-                }
-            }
-            if (direction == "OUTBOUND") {
-                return outboundMovementLocations
-            }
-        } else {
-            if (direction == "INBOUND") {
-                return locations.findAll {
-                    it.locationType.locationTypeCode == LocationTypeCode.SUPPLIER || !it.supports(ActivityCode.MANAGE_INVENTORY)
-                }
+        if (direction == "INBOUND") {
```

Comment:
LGTM, but I think we can improve the code (I know it was done like that previously), maybe you can use the StockMovemntDirection enum?